python/sglang/srt/layers/attention/flashattention_backend.py

```
@@ -117,23 +122,28 @@ def forward_extend(
         forward_batch: ForwardBatch,
         save_kv_cache=True,
     ):
-        cache_loc = (
-            forward_batch.out_cache_loc
-            if not layer.is_cross_attention
-            else forward_batch.encoder_out_cache_loc
-        )
 
-        if k is not None:
```

maybe keep this logic?

I thought about the same code refactor like you did, but I think it's better to assert and fail early if v is not none in any case...

---

resolved